README.md

```diff
@@ -1,15 +1,55 @@
 <a href="https://codeclimate.com/github/ASAAR/SE2-KaggleComp"><img src="https://codeclimate.com/github/ASAAR/SE2-KaggleComp/badges/gpa.svg" /></a> <a href="https://codeclimate.com/github/ASAAR/SE2-KaggleComp"><img src="https://codeclimate.com/github/ASAAR/SE2-KaggleComp/badges/issue_count.svg" /></a>
 
-# SE2-KaggleComp
+# SE2 Machine Learning Framework
+
 Software Engineering 2 class kaggle competition repo. We'll also be building a bit of a framework to help make doing these easier in the future.
 
-## Setup Instructions
-**You must be using a unix system for this software to work!**
+## Overview
+
+### Purpose
+To aid developers using machine learning algorithms in finding the best algorithms and optimal configurations for their specific situation. This is accomplished by recording as much information on a certain model as the developer wants, and then analyzing all the data to find which algorithms work best on a dataset and with what settings to they work best.
+
+### Features
+- Automatically applies several of a number of machine learning algorithms against the input data based on the settings it is given
+- Has a large number of tests that make sure algorithms included still run and haven't become outdated
+- Can be put in validation or application mode (train/test mode)
+- Record results from a machine learning algorithm test
+- Saves results in a Firebase Database
+- (Coming Soon) View results in a WebUI
+- (Coming Soon) Analyze data from results in the WebUI
+
+### Terminology
```

Comment:
Do we want all terminology to be on the main README? As opposed to having individual modules keep their own terminologies up to date? This might prevent conflicts later. Especially with words like *test* that might mean something different depending on the context